stock_partner_delivery_window/models/res_partner.py

```diff
@@ -71,68 +73,69 @@ def is_in_delivery_window(self, date_time):
         :return: Boolean
         """
         self.ensure_one()
-        windows = self.get_delivery_windows(date_time.weekday()).get(self.id)
-        if windows:
-            for w in windows:
-                start_time = w.get_time_window_start_time()
-                end_time = w.get_time_window_end_time()
-                if self.tz:
-                    utc_start = tz_utils.tz_to_utc_time(self.tz, start_time)
-                    utc_end = tz_utils.tz_to_utc_time(self.tz, end_time)
-                else:
-                    utc_start = start_time
-                    utc_end = end_time
-                if utc_start <= date_time.time() < utc_end:
-                    return True
+        if self.delivery_time_preference == "workdays":
+            if date_time.weekday() > 4:
+                return False
+            return True
+        for w in self.get_delivery_windows(date_time.weekday()):
+            start_time = w.get_time_window_start_time()
+            end_time = w.get_time_window_end_time()
+            if self.tz:
+                utc_start = tz_utils.tz_to_utc_time(self.tz, start_time)
+                utc_end = tz_utils.tz_to_utc_time(self.tz, end_time)
+            else:
+                utc_start = start_time
+                utc_end = end_time
+            if utc_start <= date_time.time() < utc_end:
+                return True
         return False
 
     def _get_delivery_time_format_string(self):
-        return _("From %s to %s")
+        return _("From {} to {}")
```

Comment:
/!\ same, this method could have been customized locally, better to not change this.